pkg/resource/plugin/analyzer_plugin.go

```diff
@@ -305,3 +313,46 @@ func convertDiagnostics(protoDiagnostics []*pulumirpc.AnalyzeDiagnostic) ([]Anal
 
 	return diagnostics, nil
 }
+
+func constructEnv(opts *PolicyAnalyzerOptions) ([]string, error) {
```

Comment:
A doc comment here would be super helpful, as TBH it isn't clear what the slice being returned actually is. Key value pairs to export environment variables required for the policy pack?

Reply:
Sure. We have existing variants of this code for each of the language hosts, for example:

https://github.com/pulumi/pulumi/blob/048acc24f774fa51916d418c45ccc31a05f3f250/sdk/dotnet/cmd/pulumi-language-dotnet/main.go#L512-L548

Reply:
Ah interesting. That's really good to know, and certainly worth calling that out in the doc comment ;)